### DISCUSSION TITLE
Projecting a vector layer onto 3D terrain?

### DISCUSSION THREAD

**Discussion by diehl** [opengeos/leafmap#897] [Q&A] [answered]
Inspired by the 3D terrain example [here](https://leafmap.org/maplibre/3d_terrain/), where OSM raster tiles are projected onto terrain data, I'm wondering if it is possible to do something analogous with a vector layer? Does MapLibre GL JS support that?

**Comment by giswqs** [comment, marked as answer]
If you enable the 3D basemap, all layers added later should be automatically protected onto the 3D terrain automatically by MapLibre.

Alternatively, if the vector layer has a height attribute, you can use the fill-extrusion layer 

https://leafmap.org/maplibre/3d_choropleth/

**Reply by diehl** [reply]
Thank you @giswqs for your prompt reply!
